### ISSUE TITLE
Feature Request: Integrate Digital Ocean's API for Floating API

### ISSUE THREAD

**Issue by benyanke** [picluster/picluster#91]
When running on Digital Ocean VPS instances, the typical floating IP management won't work properly.

As @rusher81572 mentioned [here](https://github.com/picluster/picluster/issues/90#issuecomment-330112780):
```
Lines 153-154 of agent.js can be customizable via an override from config.json or directly in the code with the curl command or API calls for DO. The latter would be the quicker solution in the short term.
```

Given a Digital Ocean API key, a great feature would be using DO's floating IP feature, and using the DO REST API to move the floating IP between nodes.

_(branched off from #90)_

**Comment by rusher81572** [comment]
Committed code today to add this feature. 

New args in config.json for this to work:

  "vip": [{
      "node": "192.168.0.101",
    ...
      "instance_id": "123456"  <--DO Droplet ID
    },

  "vip_provider": "digitalocean",
  "vip_api_token": "1234567890",

Just needs testing now.

**Comment by rusher81572** [comment]
Digital Ocean does not let you see the Floating IP at the OS level so this will need a major work around or unable to do it

**Comment by benyanke** [comment]
Yes, it's done on a network level - the traffic is redirected/rewritten to go to the nic on the VM, the OS doesn't actually bind an interface to the IP.

**Comment by rusher81572** [comment]
Thanks for the clarification. After learning more and failing with the code implementation, when using cloud providers, I think it is best to point DNS to a Cloud provider LB which then points to a nginx container LB running on PiCluster. Cloud providers do not make it easy to manage VIP's like traditional on-prem systems.
